dot/state/storage_test.go

```diff
@@ -184,7 +184,7 @@ func TestGetStorageChildAndGetStorageFromChild(t *testing.T) {
 		StorageValue: []byte{3, 4},
 		Dirty:        true,
 	}
-	testChildTrie := trie.NewTrie(trieRoot)
+	testChildTrie := trie.NewTrie(trieRoot, nil)
```

Comment:
maybe it's better to use a mock implementation of `db.Database` here.  This would panic if the test tries to fetch from a `nil` db.